src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -9,28 +9,54 @@ namespace Microsoft.Docs.Build
 {
     internal class MonikersProvider
     {
-        private readonly List<(Func<string, bool> glob, List<string> monikers)> _rules = new List<(Func<string, bool>, List<string>)>();
+        private readonly List<(Func<string, bool> glob, (string monikerRange, List<string> monikers))> _rules = new List<(Func<string, bool>, (string, List<string>))>();
 
         public MonikersProvider(Config config, MonikerRangeParser monikerRangeParser)
         {
             foreach (var (key, monikerRange) in config.MonikerRange)
             {
-                _rules.Add((GlobUtility.CreateGlobMatcher(key), monikerRangeParser.Parse(monikerRange)));
+                _rules.Add((GlobUtility.CreateGlobMatcher(key), (monikerRange, monikerRangeParser.Parse(monikerRange))));
             }
             _rules.Reverse();
         }
 
-        public List<string> GetMonikers(Document file)
+        public (Error, List<string>) GetMonikers(Document file, string fileMonikerRange = null)
```

Comment:
>Error [](start = 16, length = 5)

why not keep the consistency to return `List<Error>`?

Reply:
Because it's just one error, think we should not use a list if it is just one